crates/pop-cli/src/deployment_api.rs

```diff
@@ -0,0 +1,348 @@
+		let body = res.text().await?;
+		if !status.is_success() {
+			return Err(anyhow::anyhow!("Deployment failed with status {}: {}", status, body));
+		}
+		Ok(serde_json::from_str(&body)?)
+	}
+
+	/// Retrieves collator keys for a specified parachain.
+	///
+	/// # Arguments
+	/// * `id` - The ID for which collator keys are being fetched.
+	pub async fn get_collator_keys(&self, id: u32) -> Result<CollatorKeysResponse> {
+		let url = format!(
+			"{}{}",
+			self.base_url,
+			self.provider.get_collator_keys_path(&self.relay_chain_name, id)
+		);
+		let res = self
+			.client
+			.get(&url)
+			.header("Authorization", format!("Bearer {}", self.api_key))
+			.send()
+			.await?
+			.error_for_status()?;
+		Ok(res.json().await?)
+	}
+}
+
+/// Request payload for the deployment call.
+#[derive(Debug, Serialize)]
+pub struct DeployRequest {
+	/// The name of the chain to be deployed.
+	pub name: String,
+	/// The key of the proxy owner.
+	pub proxy_key: String,
+	/// The runtime template used.
+	pub runtime_template: Option<String>,
+	/// Sudo account for the created parachain (SS58 format).
+	pub sudo_key: String,
+	/// Collator file ID.
+	pub collator_file_id: String,
+	/// Chain specification JSON file (limited to 10 MB).
+	pub chainspec: PathBuf,
+}
+impl DeployRequest {
+	// Creates a new `DeployRequest` by parsing the chain specification file.
+	pub fn new(
+		collator_file_id: String,
+		genesis_artifacts: GenesisArtifacts,
+		proxy_address: Option<String>,
+	) -> anyhow::Result<Self> {
+		let chain_spec = ChainSpec::from(&genesis_artifacts.chain_spec)?;
+		let chain_name = chain_spec
+			.get_name()
+			.ok_or_else(|| anyhow::anyhow!("Failed to retrieve chain name from the chain spec"))?;
+		let proxy_key = proxy_address
+			.as_deref()
+			.map(|s| s.trim_start_matches("Id(").trim_end_matches(")"))
+			.unwrap_or("")
+			.to_string();
+		let template = chain_spec
+			.get_property_based_on()
+			.and_then(Parachain::deployment_name_from_based_on);
+		let sudo_address = chain_spec.get_sudo_key().ok_or_else(|| {
+			anyhow::anyhow!("Failed to retrieve the sudo address from the chain spec")
+		})?;
+		Ok(Self {
+			name: chain_name.to_string(),
+			proxy_key,
+			runtime_template: template,
+			sudo_key: sudo_address.to_string(),
+			collator_file_id,
+			chainspec: genesis_artifacts.raw_chain_spec,
+		})
+	}
+}
+
+/// Response from the deployment call.
+#[derive(Debug, Deserialize)]
+pub struct DeployResponse {
+	pub status: String,
+	#[serde(rename = "rollupUrl")]
+	pub message: String,
+}
+
+/// Response from a collator keys request.
+#[derive(Debug, Deserialize)]
+pub struct CollatorKeysResponse {
+	/// Collator file ID.
+	#[serde(rename = "fileId")]
+	pub collator_file_id: String,
+	/// List of the collator keys.
+	#[serde(rename = "publicCollatorKey", deserialize_with = "deserialize_single_or_vec")]
+	pub collator_keys: Vec<String>,
+}
+fn deserialize_single_or_vec<'de, D>(deserializer: D) -> Result<Vec<String>, D::Error>
```

Comment:
Currently, the API sends only one collator key, but ideally, it should support multiple. Until the API is updated, this method parses the response into a `Vec<String>` to accommodate future changes.